application/comit_node/src/bam_api/rfc003/swap.rs

```diff
@@ -68,6 +72,30 @@ pub fn swap_config(
                         Box::new(response_receiver.then(move |result| {
                             match result {
                                 Ok(rfc003::bob::SwapResponseKind::BitcoinEthereum(response)) => Ok(to_bam_response::<Bitcoin, Ethereum>(response)),
+                                Ok(invalid_response) => {
+                                    error!("Unexpected response kind {:?} for swap {}", invalid_response, swap_id);
+                                    Ok(Response::new(Status::RE(0)))
+                                }
+                                Err(_) => {
+                                    warn!("Failed to receive from oneshot channel for swap {}", swap_id);
+                                    Ok(Response::new(Status::SE(0)))
```

Comment:
If it's a fail on Bob side (and we are Bob here, right?) then it should be `RE(0)`

Reply:
We are Bob here, yeah, but I am not sure about this one. This is just what we were returning in this case before this PR.

Reply:
Yes what we were returning is incorrect. However, I will be opening an issue as part of the RFC to fix it so happy to ignore.